PrebidMobile/PrebidMobile-core/src/main/java/org/prebid/mobile/PrebidNativeAd.java

```diff
@@ -309,6 +309,18 @@ public void onImpressionTrackerFired() {
                     });
                     impressionTrackers.add(impressionTracker);
                 }
+            } else if (impEvent != null){
```

Comment:
Here are some things that should be improved here:
- we shouldn't use the `else if`. From some perspective, it will fix the issue, but it makes the logic more confusing. The `impEvent` is an independent property that doesn't relate to event trackers. So, it would be incorrect to process them separately only if event trackers are empty. 
- creating the `ImpressionTracker.create(impEvent, ...` we will track the event twice. Once inside the ImoressionTracker, and then in `notifyImpressionEvent()`.
- there is one more registerView method in this class that copies, 😔, the same tracking algorithm and the issue. So we should implement the generic solution for both methods without copying the code.   

I'd suggest to implement it in the following way:
- merge `imp_trackers` and `impEvent` into temp array
- for each entry of the temp array create respective `ImpressionTracker`
- in the `onImpressionTrackerFired` make sure that the `listener.onAdImpression();` will be called only once
- remove `notifyImpressionEvent`, but be sure that the impEvent will be tracked by respective `ImpressionTracker`
- wrap this approach into internal function and call it in both `registerView` methods